js/reference.js

```diff
@@ -4710,6 +4987,30 @@
         }
         return res;
     };
+    
+    /**
+     * Represents entity filters that can be applied to facet.
+     * 
+     * @param       {ERMrest.tuple} tuple the tuple object
+     * @constructor
+     */
+    function EntityFacetFilter(tuple, col) {
+        
+        ChoiceFacetFilter.superClass.call(this, tuple._data);
+        this.tuple = tuple;
+        this.facetFilterKey = "choices";
+    }
+    module._extends(EntityFacetFilter, FacetFilter);
+    
+    /**
+     * String representation of entity filter. It will be the tuple displayname
+     * 
+     * @return {string}
+     */
+    EntityFacetFilter.prototype.toString = function () {
+        //TODO: should it be unformatted or not?
```

Comment:
Yeah exactly. Most of these TODOs for me to not forget to change it when I get to actually use it in chaise. That was one of the reasons that I didn't wanted to create this PR. There are a bunch for TODOs that should be addressed but cannot be addressed until we get to integrate it with chaise.